301-quarkus-vertx-kafka/src/test/java/io/quarkus/qe/vertx/kafka/QuarkusVertxKafkaResource.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.qe.vertx.kafka;
+
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.lifecycle.Startables;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public class QuarkusVertxKafkaResource implements QuarkusTestResourceLifecycleManager {
+
+    KafkaContainer kafkaContainer;
+    SchemaRegistryContainer schemaRegistry;
+    Map<String, String> config = new HashMap<>();
+
+    @Override
+    public Map<String, String> start() {
+        Network network = Network.newNetwork();
+        kafkaContainer = new KafkaContainer("5.3.0").withNetwork(network);
+        schemaRegistry = new SchemaRegistryContainer("5.3.0").withNetwork(network).withKafka(kafkaContainer);
+
+        Startables.deepStart(Stream.of(kafkaContainer, schemaRegistry)).join();
+
+        String kafkaUrl = kafkaContainer.getBootstrapServers();
+        System.out.println(kafkaUrl);
+        String registryUrl = schemaRegistry.getSchemaRegistryUrl();
+        System.out.println(registryUrl);
+
+
+        config.put("kafka.bootstrap.servers", kafkaUrl);
+        config.put("mp.messaging.connector.smallrye-kafka.schema.registry.url", registryUrl);
+
+        config.put("mp.messaging.outgoing.source-stock-price.schema.registry.url", registryUrl);
+        config.put("mp.messaging.incoming.channel-stock-price.schema.registry.url", registryUrl);
+        config.put("mp.messaging.outgoing.sink-stock-price.schema.registry.url", registryUrl);
+        config.put("mp.messaging.incoming.test-sink-stock-price.registry.url", registryUrl);
+
+        return config;
+    }
+
+    @Override
+    public void stop() {
+        kafkaContainer.stop();
+        schemaRegistry.stop();
```

Comment:
Any of these instances might be null (if something fails before running the start method).